### ISSUE TITLE
Provide a helper to create/connect to a signal directly

### ISSUE THREAD

**Issue by bilelmoussaoui** [gtk-rs/gtk-rs#360]
Recently, in gtk4-rs we added bindings for `gtk_widget_class_install_action` which works like this 
```rust
klass.install_action(
    "win.open",
    None,
    move |win, _action_name, _action_target| {
        let self_ = imp::VideoPlayerWindow::from_instance(&win);
        self_.dialog.set_transient_for(Some(win));
        self_
            .dialog
            .connect_response(clone!(@weak win => move |d, response| {
                if response == gtk::ResponseType::Accept {
                    win.set_video(d.get_file().unwrap());
                }
                d.destroy();
            }));

        self_.dialog.show();
    },
);
```
It would be interesting to have something similar but for signals

**Comment by sdroege** [comment]
I don't understand what you mean with this issue :) Can you explain in more detail?

**Comment by bilelmoussaoui** [comment]
> I don't understand what you mean with this issue :) Can you explain in more detail?

Actually, I am still not sure how to make something useful out of this, often you connect the signals on other objects than the one you're currently creating. There's also `connect_` part where we have to be careful with whether we want to enforce Send trait or not depending on the use cases. Did you had something specific in mind?

**Comment by sdroege** [comment]
Me? You suggested this here :)

What I suggested in the PR was about making `install_action()` from something that the user calls into a function on the `Impl` trait like we do for signals/etc. And that should probably come with a builder for the actions, similar to the one for signals. Is that what you mean here?

**Comment by bilelmoussaoui** [comment]
Oh, I completely mis-understood your other comment then. I'm not sure if we want to provide a Builder pattern for creating actions, it could be useful but I think something like #240 should cover more than enough use cases.

The `install_action` itself is useful as you don't need to have your widget implementing `ActionMap` interface to add the actions to it which makes it much more interesting. 

I will update the issue to reflect that :)

**Comment by sdroege** [comment]
> The `install_action` itself is useful as you don't need to have your widget implementing `ActionMap` interface to add the actions to it which makes it much more interesting.

It would be part of `WidgetImpl` then or not? :)

**Comment by bilelmoussaoui** [comment]
Not really, as you can use can define actions on any object. I guess as part of a new trait exposed by `gio` itself, so maybe a new `ActionsImpl`?  Then maybe #240 could just generate such trait for you.

**Comment by sdroege** [comment]
On any object? How does that work? :)

**Comment by bilelmoussaoui** [comment]
The actions themselves can be anywhere yes, it depends on the use cases of course. For the actions to be useful, they have to be added to an action map with `ActionMapExt::add_action`. It could be a widget, or just a `gio::Application` in case you're writing a CLI app, a DBus server or whatever.

In GTK only two objects implement such interface, `gtk::Application` with the action group name `app` and  `gtk::ApplicationWindow` with `win` as an action group name. If you want to use your actions on your new widget, you either have to implement the interface manually or use a `gio::SimpleActionGroup`. The custom action group can be added to a widget using `WidgetExt::insert_action_group` so that GTK's internals know where to look for actions.

In the case for say a CLI app, you can look for an action by it's name form your action group and call activate on it.
